genie-web/src/main/java/com/netflix/genie/web/controllers/JobSearchAspect.java

```diff
@@ -0,0 +1,122 @@
+            h = (Health) joinPoint.proceed(joinPoint.getArgs());
+            tags.put(MetricsConstants.TagKeys.STATUS, h.getStatus().toString());
+        } catch (final Throwable t) {
+            MetricsUtils.addFailureTagsWithException(tags, t);
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            log.debug("HealthEndpoint.invoke() completed in {} ns", turnaround);
+            registry
+                .timer(registry.createId("health-check.endpoint.timer", tags))
+                .record(turnaround, TimeUnit.NANOSECONDS);
+        }
+        return h;
+    }
+
+    @Around(
+        "execution("
+            + "  org.springframework.boot.actuate.health.Health "
+            + "  org.springframework.boot.actuate.health.HealthIndicator.health()"
+            + ")"
+    )
+    public Health healthIndicatorHealthMonitor(ProceedingJoinPoint joinPoint) throws Throwable {
+        final long start = System.nanoTime();
+        final Health h;
+        Throwable throwable = null;
+        try {
+            h = (Health) joinPoint.proceed(joinPoint.getArgs());
+        } catch (Throwable t) {
+            throwable = t;
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            recordHealthIndicatorTurnaround(turnaround, joinPoint, throwable);
+        }
+        return h;
+    }
+
+    @Around(
+        "execution("
+            + "  void " +
+            "    org.springframework.boot.actuate.health.AbstractHealthIndicator.doHealthCheck("
+            + "    org.springframework.boot.actuate.health.Health.Builder"
+            + "  )"
+            + ")"
+    )
+    public void AbstractHealthIndicatorDoHealthCheckMonitor(ProceedingJoinPoint joinPoint) throws Throwable {
+        final long start = System.nanoTime();
+        Throwable throwable = null;
+        try {
+            joinPoint.proceed(joinPoint.getArgs());
+        } catch (Throwable t) {
+            throwable = t;
+            throw t;
+        } finally {
+            final long turnaround = System.nanoTime() - start;
+            recordHealthIndicatorTurnaround(turnaround, joinPoint, throwable);
+        }
+    }
+
+    private void recordHealthIndicatorTurnaround(
+        long turnaround,
+        ProceedingJoinPoint joinPoint,
+        @Nullable Throwable throwable
+    ) {
+        log.debug(
+            "{} completed in {} ns (exception: {})",
+            joinPoint.getTarget().getClass().getSimpleName(),
+            turnaround,
+            throwable != null ? throwable.getClass().getSimpleName() : "none"
+        );
+        final Map<String, String> tags;
+        if (throwable == null) {
+             tags = MetricsUtils.newSuccessTagsMap();
+        } else {
+            tags = MetricsUtils.newFailureTagsMapForException(throwable);
+        }
+        tags.put(HEALTH_INDICATOR_NAME, joinPoint.getTarget().getClass().getSimpleName());
```

Comment:
For this make sure we're not getting back one of the Spring wrapper classes as the names. Sometimes you need to cast it to the true class the spring aspect class is wrapping.

Reply:
From local testing it looks right (i.e. no need for ClassAware)